src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte

```diff
@@ -4,14 +4,81 @@
     import { page } from '$app/stores';
     import { Submit, trackError, trackEvent } from '$lib/actions/analytics';
     import { Modal } from '$lib/components';
-    import { Button } from '$lib/elements/forms';
+    import { Button, FormList, InputText } from '$lib/elements/forms';
     import { addNotification } from '$lib/stores/notifications';
     import { sdk } from '$lib/stores/sdk';
     import { database } from './store';
+    import {
+        TableBody,
+        TableCellHead,
+        TableHeader,
+        TableCell,
+        TableRow,
+        TableScroll
+    } from '$lib/elements/table';
+    import { toLocaleDate } from '$lib/helpers/date';
+    import { type Models, Query } from '@appwrite.io/console';
     const databaseId = $page.params.database;
 
     export let showDelete = false;
 
+    let error = null;
+    let databaseName: string = null;
+    let isLoadingDocumentsCount = false;
+
+    /* enable overflow-x */
+    const columnWidth = 135;
+
+    $: collectionItems = [];
+    let collections: Models.CollectionList = null;
+
+    function buildQueries(): string[] {
+        const queries = [Query.orderDesc('$updatedAt')];
+
+        if (collectionItems.length > 0) {
+            queries.push(Query.limit(25));
+            queries.push(Query.offset(collectionItems.length));
+        } else {
+            queries.push(Query.limit(3));
+        }
+
+        return queries;
+    }
+
+    async function listCollections() {
+        // let's just wait...
+        if (isLoadingDocumentsCount) return;
+
+        isLoadingDocumentsCount = true;
+
+        try {
+            const queries = buildQueries();
+
+            collections = await sdk.forProject.databases.listCollections(databaseId, queries);
+
+            const collectionPromises = collections.collections.map(async (collection) => {
+                const { total: totalDocuments } = await sdk.forProject.databases.listDocuments(
+                    databaseId,
+                    collection.$id,
+                    [Query.limit(1)] /* limit 1, just to get the total count */
+                );
+
+                return {
+                    id: collection.$id,
+                    name: collection.name,
+                    count: totalDocuments,
+                    updatedAt: collection.$updatedAt
+                };
+            });
+
+            collectionItems = [...collectionItems, ...(await Promise.all(collectionPromises))];
+        } catch (err) {
+            error = true;
+        } finally {
+            isLoadingDocumentsCount = false;
+        }
+    }
```

Comment:
I am not sure we can do this, there can be an infinite amount of collections - therefore we should probably implement backend for this.

Reply:
But wont the queries help? This is sort of paginated behaviour. wdyt?

Reply:
Done.